googleapiclient/http.py

```diff
@@ -56,11 +58,12 @@
 from googleapiclient.model import JsonModel
 from oauth2client import util
 
-
 DEFAULT_CHUNK_SIZE = 512*1024
 
 MAX_URI_LENGTH = 2048
 
+_TOO_MANY_REQUESTS = 429
+
 
 def _retry_request(http, num_retries, req_type, sleep, rand, uri, method, *args,
```

Comment:
I could move the logic to decide what to do based on `resp.status` and `content`, but I think this would be less readable overall. If you think this is better I'm happy to do it.